src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,83 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfRequired();
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        for (Node<K, V> node = table[index]; node != null;
+                node = node.next) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node(key, value);
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        for (Node<K, V> currentNode = table[index]; currentNode != null;
+                currentNode = currentNode.next) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null ? 0 : ((int)(Math.pow(key.hashCode(), 2)))) % table.length;
```

Comment:
```suggestion
        return (key == null ? 0 : ((int)(Math.pow(key.hashCode(), 2)))) % table.length;
```
completixy of hash function is key to overall complexity of standard collection's operations. Math.pow increases complexity. You probably did it to avoid having negative values. Use Math.abs() instead, or apply a bitmask like in original impl